cosmos/gaia
# test: add e2e test for ibc bypass msg

## Description
Test ibc bypass msgs in e2e test:
### Setup for tests:
- Two chains both require transactions fees.
- hermes relayer that have two config files, one allows paying  transaction fees, one not.
- The bypass-msg setups of the two chains are:

```
  "bypass_min_fee_msg_types": [
    "/ibc.core.channel.v1.MsgRecvPacket",
    "/ibc.core.channel.v1.MsgAcknowledgement",
    "/ibc.core.client.v1.MsgUpdateClient",
  ],
  "max_total_bypass_minFee_msg_gas_usage": "1000000"
}
```
### Tests:
- Send a transaction with bypass_min_fee_msgs_types and gas consumption is within max_total_bypass_minFee_msg_gas_usage, this transaction should be free.

- Send a transaction with mixing bypass_min_fee_msgs_types and non-bypass-msgs-types without paying fees, this transaction should fail.

- Send transitions with bypass_min_fee_msgs_types and with gas consumption higher than max_total_bypass_minFee_msg_gas_usage, this transaction should fail.


Closes: #https://github.com/cosmos/gaia/issues/2486





---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...
